src/vanilla/internals.ts

```diff
@@ -838,7 +838,11 @@ const buildStore = (
   const unmountAtom =
     buildingBlockFunctions[8] ||
     ((atom) => {
-      const atomState = ensureAtomState(atom)
```

Comment:
I think we want to keep this, as this is the single point of accessing the atomStateMap. There's possibility to extend it, and I don't feel like adding more lines for this.

Reply:
@dai-shi 
Got it! I thought it could be intentional too.

Maybe it's a small overhead, but I thought it was causing unnecessary work right now, and as a beginner, I couldn't determine if atomOnInit hooks might cause any unintended problems or not.

That seems pretty reasonable if it's just for the purpose of ensuring and adhering to the converntions.

Thank you for your reply!

Reply:
Yeah, for the normal use case, atom state should have been created in advance.